drivers/soundwire/intel_init.c

```diff
@@ -263,6 +267,25 @@ static struct sdw_intel_ctx
 		md->driver->probe(md, link);
 
 		list_add_tail(&link->list, &ctx->link_list);
+		bus = &link->cdns->bus;
+		/* Calculate number of slaves */
+		list_for_each_entry(slave, &bus->slaves, node)
+			num_slaves++;
+	}
+
+	ctx->ids = kcalloc(num_slaves, sizeof(*ctx->ids), GFP_KERNEL);
```

Comment:
where will this be freed?

Reply:
devm_kcalloc ?

Reply:
I'd rather add it to `sdw_intel_cleanup()`

Reply:
Yes remove it on ``sdw_intel_cleanup()`` is better.

Reply:
btw. I think the order of error labels in ``sdw_intel_probe_controller()`` is odd. Please see #1674

Reply:
correct, this needs to be fixed as well. Nice catch

Reply:
@ranj063 @RanderWang @lyakh @plbossart Added the free in sdw_intel_cleanup()